src/azure-cli/azure/cli/command_modules/acs/addonconfiguration.py

```diff
@@ -0,0 +1,656 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import datetime
+import json
+
+from azure.cli.core.azclierror import (
+    AzCLIError,
+    ClientRequestError,
+)
+from azure.cli.core.commands import LongRunningOperation
+from azure.cli.core.commands.client_factory import (
+    get_mgmt_service_client,
+)
+from azure.cli.core.profiles import ResourceType
+from azure.cli.core.util import sdk_no_wait, send_raw_request
+from azure.core.exceptions import HttpResponseError
+from knack.log import get_logger
+from msrestazure.azure_exceptions import CloudError
+from msrestazure.tools import parse_resource_id, resource_id
+
+from ._client_factory import cf_resource_groups, cf_resources
+from ._consts import (
+    CONST_INGRESS_APPGW_ADDON_NAME,
+    CONST_INGRESS_APPGW_APPLICATION_GATEWAY_ID,
+    CONST_INGRESS_APPGW_SUBNET_CIDR,
+    CONST_INGRESS_APPGW_SUBNET_ID,
+    CONST_MONITORING_ADDON_NAME,
+    CONST_MONITORING_LOG_ANALYTICS_WORKSPACE_RESOURCE_ID,
+    CONST_VIRTUAL_NODE_ADDON_NAME,
+)
+from ._resourcegroup import get_rg_location
+from ._roleassignments import add_role_assignment
```

Comment:
The signature of the function `add_role_assignment` is different from that in aks-preview. We pass `cmd` instead of `cli_ctx` (included in the `cmd` object) to add support for backward compatibility when creating the role assignment. In an older version of api, we may not have certain model. For details, may refer the previous [PR #20376](https://github.com/Azure/azure-cli/pull/20376/files#diff-c65fefebd9d60a7e189ec65eecfae48356fab34272b2f6888c8ec21123d0db38R130-R140).